### ISSUE TITLE
bug: False positives when cd'ing with Bats-specific variables

### ISSUE THREAD

**Issue by hyperupcall** [koalaman/shellcheck#2552]
#### For bugs
- Rule Id (if any, e.g. SC2164): 
- My shellcheck version (`shellcheck --version` or "online"): 

```sh
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.7.2
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
# shellcheck shell=bash

setup() {
	cd "$BATS_TEST_TMPDIR"
}

teardown() {
	cd "$BATS_SUITE_TMPDIR"
}


```

#### Here's what shellcheck currently says:

```txt
Use 'cd ... || exit' or 'cd ... || return' in case cd fails
```

#### Here's what I wanted or expected to see:

Nothing

In Bats, `set -e` is set, so when `cd` fails, it will abort the process and show a stacktrace. When Bats [special variables](https://bats-core.readthedocs.io/en/stable/writing-tests.html#special-variables) like `BATS_SUITE_TMPDIR` are used, I think it would be nice for that to imply that the current file is running in the context of Bats, so the rule `SC2086` should be ignored

**Comment by koalaman** [comment]
Matching on a specific variable name is somewhat fragile. Have you considered explicitly adding a `set -e` to the script since you're already assuming it's set? This would also silence the suggestion.

**Comment by hyperupcall** [comment]
Yeah I did - but that would make the code harder to reason about in my case. A lot of my Bash code has shellcheck comments to hide the false positives anyways, and this will only add to the noise. I have avoided adding file level ignores for the same reason, but I suppose I should reconsider, given that this feature is unlikely to be implemented

ps . Thank you for developing ShellCheck!
